packages/features/ee/teams/api/upgrade.ts

```diff
@@ -39,22 +39,25 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (!team) {
     const prevTeam = await prisma.team.findFirstOrThrow({ where: { id } });
     const metadata = teamMetadataSchema.parse(prevTeam.metadata);
-    if (!metadata?.requestedSlug) throw new HttpError({ statusCode: 400, message: "Missing requestedSlug" });
-
+    /** Legacy teams already have a slug, this will allow them to upgrade as well */
+    const slug = prevTeam.slug || metadata?.requestedSlug;
+    if (!slug) throw new HttpError({ statusCode: 400, message: "Missing team slug" });
     try {
-      team = await prisma.team.update({
+      /** We save the metadata first to prevent duplicate payments */
+      await prisma.team.update({
         where: { id },
         data: {
-          slug: metadata.requestedSlug,
           metadata: {
             paymentId: checkoutSession.id,
             subscriptionId: subscription.id || null,
             subscriptionItemId: subscription.items.data[0].id || null,
           },
         },
       });
+      /** Then we try to upgrade the slug, which may fail if a conflict came up since team creation */
+      team = await prisma.team.update({ where: { id }, data: { slug } });
     } catch (error) {
```

Comment:
Had to split the update so we don't miss saving payment data in case of slug conflict.